src/client/Client.cc

```diff
@@ -2878,7 +2881,7 @@ void Client::_handle_full_flag(int64_t pool)
 
 void Client::handle_osd_map(const MConstRef<MOSDMap>& m)
 {
-  std::scoped_lock cl(client_lock);
+  std::unique_lock cl(client_lock);
```

Comment:
you mentioned in commit message that  the async callback registered also takes the 'client_lock' which
the `handle_osd_map` takes at the beginning. Since there are multiple code paths acquiring locks, don't you think scoped lock here makes better sense?

Reply:
didnt read below code